server/tracing/tracing.go

```diff
@@ -0,0 +1,52 @@
+package tracing
+
+import (
+	"github.com/tigrisdata/tigris/server/config"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/profiler"
+)
+
+func getTracingOptions(config *config.Config) []tracer.StartOption {
+	var opts []tracer.StartOption
+	opts = append(opts, tracer.WithTraceEnabled(config.Tracing.Enabled))
+	opts = append(opts, tracer.WithService(config.Tags.Service))
+	opts = append(opts, tracer.WithEnv(config.Tags.Environment))
+	opts = append(opts, tracer.WithServiceVersion(config.Tags.Version))
+	if config.Tracing.WithUDS != "" {
+		opts = append(opts, tracer.WithUDS(config.Tracing.WithUDS))
+	}
+	if config.Tracing.WithAgentAddr != "" {
+		opts = append(opts, tracer.WithAgentAddr(config.Tracing.WithAgentAddr))
+	}
+	if config.Tracing.WithDogStatsdAddr != "" {
+		opts = append(opts, tracer.WithAgentAddr(config.Tracing.WithAgentAddr))
+	}
+
+	return opts
+}
+
+func getProfilingOptions(config *config.Config) []profiler.Option {
+	var opts []profiler.Option
+	opts = append(opts, profiler.WithService(config.Tags.Service))
+	opts = append(opts, profiler.WithEnv(config.Tags.Environment))
+	opts = append(opts, profiler.WithVersion(config.Tags.Version))
+	return opts
+}
+
+func InitTracer(config *config.Config) error {
+	if !config.Profiling.Enabled {
+		return nil
+	}
+
+	tracer.Start(getTracingOptions(config)...)
+	defer tracer.Stop()
```

Comment:
How about these tracer and profiler defers?
you probably need to pass them up to the main.
`return func() { trace.Stop(); profiler.Stop() }, nil`
and InitTracer function would return `(func(), error)`.

Reply:
actually `func() { ulog.E(trace.Stop()); ulog.E(profiler.Stop()) }`
if `Stop`s return error

Reply:
Thanks for the tip!

Reply:
yeah, this tracer needs to be returned so that it can be stopped in main.